src/components/TextField/TextField.tsx

```diff
@@ -41,13 +44,16 @@ export const TextField: React.FC<Props> = ({
         <input
           type="text"
           id={id}
+          name={name}
           data-cy={`movie-${name}`}
           className={classNames('input', {
             'is-danger': hasError,
           })}
           placeholder={placeholder}
           value={value}
-          onChange={event => onChange(event.target.value)}
+          onChange={event => onChange(
+            { ...movie, [event.target.name]: event.target.value },
+          )}
```

Comment:
it is bad practice to make anonymous arrow functions in jsx.
instead make handler in jsx and pass reference to jsx.
destructure event.target.
use prevMovie state.
```
const handleChange = (event: React.ChangeEvent<HtmlInputElement>) => {
   const { name, value } = event.target;
   onChange( prevMovie =>  ({ ...prevMovie, [name]: value, });
}
```

```suggestion
          onChange={handleChange}
```